controller/engine_controller.go

```diff
@@ -1735,19 +1740,27 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 			log.Debug("Finished snapshot purge, will start rebuilding then")
 		}
 
+		// check and reset replica rebuild failed condition
+		if err := ec.updateReplicaRebuildFailedCondition(replica, ""); err != nil {
+			log.WithError(err).Errorf("Failed to update rebuild status information on replica %v", replicaName)
+			return
+		}
```

Comment:
Can we change to the below, so later we should not need to requery the replica.

```go
replica, err = ec.updateReplicaRebuildFailedCondition(replica, "")
if err != nil {
    log.WithError(err).Errorf("Failed to update rebuild status information on replica %v", replicaName)
    return
}
```

Reply:
Changed. Thanks